### ISSUE TITLE
publicKeyJwk, publicKeyHex, publicKeyBase64, publicKeyBase58 missing from context.

### ISSUE THREAD

**Issue by brentzundel** [w3c/did-core#23]
@OR13 moved from CCG (https://github.com/w3c-ccg/did-spec/issues/152)

**Comment by OR13** [comment]
I suggest hosting the JSON-LD Contexts and their human readable documentation under this github repo, so that contributors can open a single PR to fix both human and machine readable spec's related to DID security. 

Security and usability issues arise when documentation becomes out of date and JSON-LD context resolution spans many domains controlled by various 3rd parties.

**Comment by dmitrizagidulin** [comment]
Related: https://github.com/w3c/did-spec/issues/5

**Comment by awoie** [comment]
This issue is related to #55 . Support for `ethereumAddress` is also missing in context.

**Comment by tplooker** [comment]
Also related to #67

**Comment by selfissued** [comment]
The publicKeyHex, publicKeyBase64, and publicKeyBase58 structures are duplicative representations of the same information.  Having more than one way to do things will inevitably create interop problems, as developers will only build the ones that they think they need at present.  And these sets will differ between implementations and deployments.

It should be no surprise that I'd advocate replacing all of them with JWK, since it's already a widely-implemented JSON-based standard.  That would increase interoperability and reduce unnecessary code fragmentation and bloat.

**Comment by msporny** [comment]
> I suggest hosting the JSON-LD Contexts and their human readable documentation under this github repo

Tagging @OR13 since he could close this issue out by putting in a PR for an experimental `@context` hosted in the /contexts/ directory of this repo.

**Comment by OR13** [comment]
I'm a bit confused about what goes in what repo now... shouldn't the context and schema definitions got in the registry repo?

**Comment by OR13** [comment]
Blocked by: https://github.com/w3c/did-core-registry/issues/3

**Comment by OR13** [comment]
Blocked by: https://github.com/w3c/did-core-registries/pull/20

**Comment by OR13** [comment]
JWK is unblocked: https://github.com/w3c/did-spec-registries/blob/master/contexts/did-v1.jsonld#L119

**Comment by OR13** [comment]
publicKeyHex blocked, but we will probably just define that in the next did spec registries updates.

**Comment by rhiaro** [comment]
I'm not sure of the consensus around which of these will be defined in the DID Core spec. I think publicKeyJwk and publicKeyBase58 for sure? Any not in the Core Spec will be added to the Registries (linking to external specs and contexts). I can PR to add the agreed ones to the Core context and to the spec (#281) when there's a definitive answer.

**Comment by msporny** [comment]
> I think publicKeyJwk and publicKeyBase58 for sure?

Correct.

The only issue with publicKeyBase58 might be pointing to a normative spec. We can get around that by not making any normative statements on what the expected encoding is. We may want to try to say something like using the "Base58 Bitcoin alphabet"... or referring to the I-D at IETF.

> Any not in the Core Spec will be added to the Registries (linking to external specs and contexts).

Yep, exactly.

**Comment by kdenhartog** [comment]
I found that publicKeyHex was already defined in the [did-spec-registries](https://w3c.github.io/did-spec-registries/#publickeyhex), so wasn't sure what the best approach for context definitions would be. Looking to get it added to the [CCG security-vocab](https://github.com/w3c-ccg/security-vocab/pull/40) as well soon.

**Comment by OR13** [comment]
Yes, we need `publicKeyHex` to be defined in security-vocab jsonld and the html documentation which the jsonld contexts reference.. we have an open PR for the HTML we need context update in the W3C CCG

**Comment by OR13** [comment]
Suggest closing, we have this in the spec registries.

**Comment by OR13** [comment]
recommend opening a new issue remove 'publicKeyPem`.

**Comment by OR13** [comment]
closing, opened https://github.com/w3c/did-spec-registries/issues/126
